tests/ValidateSbrpTests/ValidateSbrpTests.cs

```diff
@@ -0,0 +1,170 @@
+    }
+
+    [Fact]
+    public void CheckForSbrpAttribute()
+    {
+        string[] targetPacks = Directory.GetDirectories(Path.Combine(RepoRoot, "src/targetPacks/ILsrc")).Select(x => Path.GetFileName(x).ToLower()).ToArray();
+        string[] textOnlyPackages = Directory.GetDirectories(Path.Combine(RepoRoot, "src/textOnlyPackages/src")).Select(x => Path.GetFileName(x).ToLower()).ToArray();
+
+        var filteredPackages = new List<string>();
+        
+        foreach (var package in Packages)
+        {
+            string packageName = Path.GetFileNameWithoutExtension(package).ToLower();
+            packageName = Regex.Replace(packageName, VerionPattern, string.Empty);
+            if (!targetPacks.Contains(packageName) && !textOnlyPackages.Contains(packageName))
+            {
+                filteredPackages.Add(package);
+            }
+        }
+
+        foreach (var package in filteredPackages)
+        {
+            string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName(), Path.GetFileNameWithoutExtension(package));
+
+            try
+            {
+                Directory.CreateDirectory(tempDirectory);
+
+                ZipFile.ExtractToDirectory(package, tempDirectory);
+
+                var dlls = Directory.GetFiles(tempDirectory, "*.dll", SearchOption.AllDirectories);
+
+                foreach (var dll in dlls)
+                {
+                    using var stream = new FileStream(dll, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+                    using var peReader = new PEReader(stream);
+
+                    MetadataReader reader = peReader.GetMetadataReader();
+
+                    Assert.True(HasSbrpAttribute(reader), $"{package}/{Path.GetRelativePath(tempDirectory, dll)} does not contain the {SbrpAttributeType} attribute with key='source' and value='source-build-reference-packages'.");
+                }
+            }
+            finally
+            {
+                Directory.Delete(tempDirectory, true);
+            }
+        }
+    }
+
+    [Fact]
+    public async Task CheckForSignature()
+    {
+        var trustProviders = new List<ISignatureVerificationProvider>()
+        {
+            new SignatureTrustAndValidityVerificationProvider()
+        };
+        var verifier = new PackageSignatureVerifier(trustProviders);
+
+        var settings = SignedPackageVerifierSettings.GetDefault();
+
+        foreach (var package in Packages)
+        {
+            bool isSigned = await IsPackageSigned(package, verifier, settings);
+            Assert.False(isSigned, $"{package} is signed. Signed packages are not allowed in source-build-reference-packages.");
+        }
+    }
+
+    private bool HasSbrpAttribute(MetadataReader reader)
+    {
+        return reader.CustomAttributes.Select(attrHandle => reader.GetCustomAttribute(attrHandle)).Any(attr => IsAttributeSbrp(reader, attr));
+    }
```

Comment:
Simplify with expression-bodied method and break apart into separate lines for readability:

```suggestion
    private bool HasSbrpAttribute(MetadataReader reader) =>
        reader.CustomAttributes
            .Select(attrHandle => reader.GetCustomAttribute(attrHandle))
            .Any(attr => IsAttributeSbrp(reader, attr));
```

Reply:
Changed in https://github.com/dotnet/source-build-reference-packages/commit/ce1a70dc071c6a5adce2b2d0ca6285031ee12258